autoload/yankitute.vim

```diff
@@ -75,3 +77,11 @@ function! s:eval(results, replace) abort
     call add(a:results, eval(a:replace[2:]))
     return submatch(0)
 endfunction
+
+function! s:expand_returns(lst) abort
+  let arr = []
+  let pat = '\v%(%(\\\\)*\\)@<!%(\\r)'
+  call map(a:lst, 'split(v:val, pat)')
+  call map(copy(a:lst), 'extend(arr, v:val)')
```

Comment:
1. If you are going to be copying the list anyways, it's better to copy the first one. That way you'll leave the argument intact - which is more polite behavior for a function(even a script-local one)
2. One-liners are good for command mode, but they make actual code harder to understand. Don't use `map` for it's side effects - it's good for one liners but makes things confusing in plugin code. In Vim's particular case, I don't count modifying the list as a side-effect - but modifying a whole different list sure is.

So, overall, I'd write this code as:


```vim
for l:item in a:lst
    call extend(l:arr, split(l:item, l:pat))
endfor
```
* No needless copies
* The intent is clear